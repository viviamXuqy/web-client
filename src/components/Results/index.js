import React, { PureComponent } from 'react';
import { object, func } from 'prop-types';
import { Button, Modal, message, Card } from 'antd';
import { injectIntl } from 'react-intl';

import Sider from '../../containers/layout/Sider';

import Cicon from '../../shared/components/Cicon';

import ResultDetail from '../../shared/containers/ResultDetail';
import ResultStructSearch from '../../shared/containers/ResultStructSearch';
import ResultFeatureSearch from '../../shared/components/ResultFeatureSearch';
import ResultSearch from '../../shared/containers/ResultSearch';

import Cookies from 'js-cookie';

import Filter from './Filter';
import List from './List';

import './index.less';

class Result extends PureComponent {
  static propTypes = {
    intl: object.isRequired,
    filter: object.isRequired,
    meta: object.isRequired,
    results: object.isRequired,
    fetchResults: func.isRequired,
    updateFilter: func.isRequired,
    deleteResults: func.isRequired,
    deleteAllResults: func.isRequired,
    clearDatas: func.isRequired,
  }

  state = {
    currResult: '', // 当前操作的结果
    modalVisible: false, // 详细信息界面
    modalVisibleStructSearch: false, // 结构化检索界面
    modalVisibleFeatureSearch: false, // 以图搜图检索界面
    search: '', // 关键字
    struct: '', // 结构化以逗号分割的字符串
    target: '',
    page: 1, // 当前页
    pageSize: 8, // 当前页数量
    begin: '', // 开始时间
    end: '', // 结束时间
    taskBeginId: '', // 任务开始ID
    taskEndId: '', // 任务结束ID
    selectedKeys: [], // 选中ID
    delSize: 0, // 一个用户删除个数，用来避免最后一页删除后，还是请求的没有用的
    isReset: false, // 是否显示重置刷新按钮
    modalVisibleSearch: false, // 高级检索
    camera: '', // 摄像机列表
  }

  componentDidMount() {
    this.handleFetchResults();
  }

  componentWillUnmount() {
    const { clearDatas } = this.props;
    clearDatas('results');
  }

  handleFetchResults = () => {
    const { fetchResults, meta } = this.props;
    const {
      search, struct, target, page, pageSize, begin, end, taskBeginId, taskEndId, delSize, camera,
    } = this.state;

    let pageTemp = Math.ceil((meta.total - delSize) / pageSize);
    pageTemp = page > pageTemp ? (page - 1) : page;
    pageTemp = pageTemp <= 0 ? 1 : pageTemp;

    const filters = {
      search,
      struct,
      target,
      page: pageTemp,
      pageSize,
      begin,
      end,
      taskBeginId,
      taskEndId,
      camera,
    };

    this.updateFilter(filters);

    fetchResults();
  }

  updateFilter = values => {
    const { updateFilter } = this.props;

    updateFilter('results', values);
  }

  handleSearch = values => {
    const options = {
      search: '',
      struct: '',
      target: '',
      page: 1,
      begin: null,
      end: null,
      taskBeginId: '',
      taskEndId: '',
      isReset: true,
      selectedKeys: [],
      camera: '',
      ...values,
    };
    this.setState({
      ...options,
    }, () => {
      this.handleFetchResults();
    });
  }

  handleReset = () => {
    this.handleSearch({ isReset: false });
  }

  handlePaginationClick = page => {
    this.setState({
      page,
      selectedKeys: [],
    }, () => {
      this.handleFetchResults();
    });
  }

  showModal = record => {
    this.setState({
      currResult: record,
      modalVisible: true,
    });
  }

  hideModal = () => {
    this.setState({
      modalVisible: false,
    });
  }

  showModalStructSearch = () => {
    this.setState({
      modalVisibleStructSearch: true,
    });
  }

  hideModalStructSearch = () => {
    this.setState({
      modalVisibleStructSearch: false,
    });
  }

  showModalFeatureSearch = () => {
    this.setState({
      modalVisibleFeatureSearch: true,
    });
  }

  hideModalFeatureSearch = () => {
    this.setState({
      modalVisibleFeatureSearch: false,
    });
  }
  hideModalSearch = () => {
    this.setState({
      modalVisibleSearch: false,
    });
  }
  handleDelete = ids => {
    const { intl, results } = this.props;
    const msg = intl.formatMessage({ id: 'App.message.delete.warn' });

    this.setState({
      delSize: ids.length,
    });

    const resultsTemp = [];
    ids.forEach(id => {
      const result = results.get(id);
      if (result && !resultsTemp.some(item => item.taskId === result.taskId)) {
        resultsTemp.push(result);
      }
    });
    const title = (
      <span>
        {msg}?&nbsp;&nbsp;
        {/**
        {resultsTemp.map(item =>
          <span key={item.taskId}>任务编号：{item.taskId}&nbsp;&nbsp;
          <Tag color="orange">{item.taskName}</Tag></span>)}？
         */}
      </span>
    );

    const self = this;

    Modal.confirm({
      title,
      okType: 'danger',
      onOk() {
        self.deleteResults(ids);
      },
    });
  }

  deleteResults = async ids => {
    const { intl, deleteResults } = this.props;

    const self = this;

    const successMsg = intl.formatMessage({ id: 'App.message.delete.success' });

    const { isOk } = await deleteResults(ids);

    if (isOk) {
      self.setState({
        selectedKeys: [],
      });
      message.success(successMsg);
      self.handleFetchResults();
    }
  }

  handleSelect = id => {
    const { selectedKeys } = this.state;
    if (selectedKeys.some(idTemp => idTemp === id)) {
      selectedKeys.splice(selectedKeys.findIndex(i => i === id), 1);
    } else {
      selectedKeys.push(id);
    }

    this.setState({
      selectedKeys: selectedKeys.slice(0),
    });
  }

  handleDeleteAllResults = () => {
    const { intl } = this.props;

    const msg = intl.formatMessage({ id: 'App.message.delete.all.warn' });
    const title = (
      <span>
        {msg}?
      </span>
    );

    const self = this;

    Modal.confirm({
      title,
      okType: 'danger',
      onOk() {
        self.deleteAllResults();
      },
    });
  }

  deleteAllResults = async () => {
    const { intl, deleteAllResults, fetchResults } = this.props;

    const successMsg = intl.formatMessage({ id: 'App.message.delete.all.success' });

    const { isOk } = await deleteAllResults();

    if (isOk) {
      message.success(successMsg);
      fetchResults();
    }
  }

  // 批量删除
  handleMultiDelete = () => {
    const { selectedKeys } = this.state;

    this.handleDelete(selectedKeys);
  }

  // 高级检索
  showModalSearch = () => {
    this.setState({
      modalVisibleSearch: true,
    });
  }

  render() {
    const {
      meta, results, filter, intl: { formatMessage },
    } = this.props;
    const { selectedKeys, currResult, isReset } = this.state;
    const user = JSON.parse(Cookies.get('user'));
    return (
      <React.Fragment>
        <Sider selectedKey="results-resultsList" />
        <div className="app-page">
          <Card
            bordered={false}
            className="app-card"
          >
            <h1 className="app-page__title">
              {formatMessage({ id: 'App.results.resultList' })}
            </h1>
            <Filter
              filter={filter}
              onSearch={this.handleSearch}
            />
            <div className="app-page__actions buttons-group">
              <Button
                type="primary"
                onClick={this.showModalSearch}
              >
                <Cicon type="structuredsearc" />{formatMessage({ id: 'App.person.struct.search' })}
              </Button>
              {/* <Button
                type="primary"
                onClick={this.showModalStructSearch}
              >
                <Cicon type="structuredsearc" />{formatMessage({ id: 'App.results.structSearch' })}
              </Button>
              <Button
                type="primary"
                onClick={this.showModalFeatureSearch}
              >
                <Cicon type="search" />{formatMessage({ id: 'App.results.searchByPic' })}
              </Button> */}
              <Button
                type="warn"
                icon="primary"
                disabled={user.grade === '3' || selectedKeys.length === 0}
                onClick={this.handleMultiDelete}
              >
                <Cicon type="delete" />{formatMessage({ id: 'App.control.deleteMore' })}
              </Button>
              <Button
                type="danger"
                icon="primary"
                onClick={this.handleDeleteAllResults}
                disabled={user.grade === '3'}
              >
                <Cicon type="delete" />{formatMessage({ id: 'App.control.clear' })}
              </Button>
              <Button
                style={{ display: (isReset ? '' : 'none') }}
                type="primary"
                onClick={this.handleReset}
              >
                <Cicon type="therefresh" />{formatMessage({ id: 'App.control.reset' })}
              </Button>
            </div>
            <List
              results={results}
              meta={meta}
              onDelete={this.handleDelete}
              onSelect={this.handleSelect}
              showResultDetail={this.showModal}
              selectedKeys={selectedKeys}
              onPaginationChange={this.handlePaginationClick}
            />
          </Card>
          {this.state.modalVisible &&
            <Modal
              width={790}
              title={formatMessage({ id: 'App.results.resultDetail' })}
              visible={this.state.modalVisible}
              onCancel={this.hideModal}
              footer={null}
            >
              <ResultDetail
                currResult={currResult}
              />
            </Modal>}
          {this.state.modalVisibleStructSearch &&
            <Modal
              width={400}
              title={formatMessage({ id: 'App.results.structSearch' })}
              visible={this.state.modalVisibleStructSearch}
              onCancel={this.hideModalStructSearch}
              footer={null}
            >
              <ResultStructSearch
                hideModal={this.hideModalStructSearch}
                onSearch={this.handleSearch}
              />
            </Modal>}
          {this.state.modalVisibleFeatureSearch &&
            <Modal
              width={863}
              title={formatMessage({ id: 'App.results.searchByPic' })}
              visible={this.state.modalVisibleFeatureSearch}
              onCancel={this.hideModalFeatureSearch}
              footer={null}
            >
              <ResultFeatureSearch
                hideModal={this.hideModalFeatureSearch}
                onSearch={this.handleSearch}
              />
            </Modal>}
          {this.state.modalVisibleSearch &&
            <Modal
              width={1128}
              title={formatMessage({ id: 'App.person.struct.search' })}
              visible={this.state.modalVisibleSearch}
              onCancel={this.hideModalSearch}
              footer={null}
            >
              <ResultSearch
                hideModal={this.hideModalSearch}
                onSearch={this.handleSearch}
              />
            </Modal>}
        </div>
      </React.Fragment>
    );
  }
}


export default (injectIntl(Result));
