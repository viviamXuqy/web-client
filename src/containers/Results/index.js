import { connect } from 'react-redux';

import { fetchResults, submitResult, deleteResults, deleteAllResults } from '../../actions/results';
import { updateFilter } from '../../actions/filter';
import { clearDatas } from '../../actions/public';
import { getResults } from '../../selectors';
import Results from '../../components/Results';

const mapStateToProps = state => ({
  user: state.user,
  results: getResults(state),
  filter: state.filter.results,
  meta: state.meta.results,
});

export default connect(mapStateToProps, {
  fetchResults,
  updateFilter,
  submitResult,
  deleteResults,
  deleteAllResults,
  clearDatas,
})(Results);
