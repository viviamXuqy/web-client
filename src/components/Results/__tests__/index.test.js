import React from 'react';
import createComponentWithIntl from 'helpers/createComponentWithIntl';
import { mountWithIntl } from 'helpers/intl-enzyme-test-helper';
import Cookies from 'js-cookie';

import { Modal, message } from 'antd';

import { getResults } from 'selectors';

import Results from 'components/Results';

import configureStore from '../../../store/__tests__/configureStore.test';

const { state } = configureStore;

const spyModalConfirm = jest.spyOn(Modal, 'confirm');
const spyMessage = jest.spyOn(message, 'success');

jest.mock('shared/containers/ResultDetail', () => 'mock-ResultDetail');
jest.mock('shared/containers/ResultStructSearch', () => 'mock-ResultStructSearch');
jest.mock('containers/layout/Sider', () => 'mock-Sider');
jest.mock('shared/containers/ResultsCard', () => 'mock-ResultsCard');

const fetchResults = jest.fn();
const updateFilter = jest.fn();
const deleteResults = jest.fn(() => Promise.resolve({ response: 'success' }));
const deleteAllResults = jest.fn(() => Promise.resolve({ response: 'success' }));
const clearDatas = jest.fn();

const initProps = {
  results: getResults(state),
  filter: {
  },
  meta: {
    page: 100000000,
    pageSize: 5,
    total: 100,
  },
  fetchResults,
  updateFilter,
  deleteResults,
  deleteAllResults,
  clearDatas,
};

function initializeData() {
  // Cookies.set('user', JSON.stringify(state.user));
  const user = {
    grade: 3,
  };
  Cookies.set('user', JSON.stringify(user));
}

const setup = (props = initProps) => {
  const component = <Results {...props} />;
  const componentWithProvider = createComponentWithIntl(component);
  const wrapper = mountWithIntl(component);

  return {
    component,
    componentWithProvider,
    wrapper,
  };
};

describe('Results component', () => {
  beforeEach(() => {
    initializeData();
  });

  it('should render without errors', () => {
    const { wrapper } = setup();

    expect(wrapper).toBeTruthy();
  });

  it('should render correct things', () => {
    const { wrapper } = setup();

    expect(wrapper.find('.app-page').length).toBe(1);
  });

  it('should render correct things when isReset is true', () => {
    const { wrapper } = setup();

    const instance = wrapper.children().instance();

    instance.setState({
      isReset: true,
    });

    expect(instance.state.isReset).toBeTruthy();
  });

  it("should render right thing and don't change unexpected", () => {
    const { componentWithProvider } = setup();
    const tree = componentWithProvider.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("should call 'handleFetchResults/updateFilter/fetchResults' during componentDidMount", () => {
    const { wrapper } = setup();
    const instance = wrapper.children().instance();
    const spyHandleFetchResults = jest.spyOn(instance, 'handleFetchResults');
    const spyUpdateFilter = jest.spyOn(instance, 'updateFilter');

    instance.componentDidMount();

    expect(spyHandleFetchResults).toHaveBeenCalledTimes(1);
    expect(spyUpdateFilter).toBeCalled();
    expect(updateFilter).toBeCalled();
    expect(fetchResults).toBeCalled();
  });

  it("should call 'clearDatas' during componentWillUnmount", () => {
    const { wrapper } = setup();
    const instance = wrapper.children().instance();
    const spyWillUnmount = jest.spyOn(instance, 'componentWillUnmount');

    wrapper.unmount();

    expect(spyWillUnmount).toHaveBeenCalledTimes(1);
    expect(clearDatas).toBeCalled();
  });

  it("should call 'handleSearch' when reset button clicked", () => {
    const { wrapper } = setup();
    const instance = wrapper.children().instance();
    const spyHandleSearch = jest.spyOn(instance, 'handleSearch');

    wrapper.find('.app-page__actions').children().at(4).simulate('click');

    expect(spyHandleSearch).toBeCalled();
  });

  it("should call 'handleFetchTaskResults' when pagination click", () => {
    const { wrapper } = setup();
    const instance = wrapper.children().instance();
    const spyHandleFetchResults = jest.spyOn(instance, 'handleFetchResults');

    wrapper.find('.ant-pagination-item').at(1).simulate('click');

    expect(spyHandleFetchResults).toBeCalled();
    expect(instance.state.page).toBe(2);
  });

  it("should show result detail modal when ResultsCard 'showResultDetail' and hide modal when closed", () => {
    const { wrapper } = setup();
    const instance = wrapper.children().instance();

    wrapper.find('mock-ResultsCard').first().props().showResultDetail();

    expect(instance.state.modalVisible).toBeTruthy();

    wrapper.update();

    wrapper.find('Modal .ant-modal-close-x').simulate('click');

    expect(instance.state.modalVisible).toBeFalsy();
  });

  it('should show struct search modal when struct search button clicked and hide modal when closed', () => {
    const { wrapper } = setup();
    const instance = wrapper.children().instance();

    wrapper.find('.app-page__actions').children().at(0).simulate('click');

    expect(instance.state.modalVisibleStructSearch).toBeTruthy();

    wrapper.find('Modal .ant-modal-close-x').simulate('click');

    expect(instance.state.modalVisible).toBeFalsy();
  });

  it('should show feature search modal when feature search button clicked and hide modal when closed', () => {
    const { wrapper } = setup();
    const instance = wrapper.children().instance();

    wrapper.find('.app-page__actions').children().at(1).simulate('click');

    expect(instance.state.modalVisibleFeatureSearch).toBeTruthy();

    wrapper.find('Modal .ant-modal-close-x').simulate('click');

    expect(instance.state.modalVisible).toBeFalsy();
  });

  it("should call 'handleDelete' when ResultsCard 'onDelete'", async () => {
    const { wrapper } = setup();
    const instance = wrapper.children().instance();

    wrapper.find('mock-ResultsCard').first().props().onDelete(['1']);
    wrapper.update();

    expect(instance.state.delSize).toBe(1);
    expect(spyModalConfirm).toBeCalled();

    const spyDeleteResults = jest.spyOn(instance, 'deleteResults');

    spyDeleteResults(['1']);

    await expect(deleteResults).toBeCalledWith(['1']);

    expect(spyMessage).toBeCalledWith('成功删除');
    expect(instance.state.selectedKeys.length).toBe(0);
  });

  it("should call 'handleDelete' when ResultsCard 'onDelete' with more same args", () => {
    const { wrapper } = setup();
    const instance = wrapper.children().instance();

    wrapper.find('mock-ResultsCard').first().props().onDelete(['1', '1']);

    wrapper.update();

    expect(instance.state.delSize).toBe(2);
    expect(spyModalConfirm).toBeCalled();
  });

  it("should call 'handleSelect' when ResultsCard 'onSelect'", () => {
    const { wrapper } = setup();
    const instance = wrapper.children().instance();

    wrapper.find('mock-ResultsCard').first().props().onSelect('1');

    expect(instance.state.selectedKeys.length).toBe(1);
  });

  it("should call 'handleSelect' when ResultsCard 'onSelect' with selectedKeys has the same key", () => {
    const { wrapper } = setup();
    const instance = wrapper.children().instance();

    instance.setState({
      selectedKeys: ['1'],
    });

    wrapper.find('mock-ResultsCard').first().props().onSelect('1');

    expect(instance.state.selectedKeys.length).toBe(0);
  });

  it("should call 'handleDeleteAllResults' when clear button clicked", async () => {
    const { wrapper } = setup();
    const instance = wrapper.children().instance();
    const spyDeleteAllResults = jest.spyOn(instance, 'deleteAllResults');

    wrapper.find('.app-page__actions').children().at(3).simulate('click');

    expect(spyModalConfirm).toBeCalled();

    spyDeleteAllResults();

    await expect(deleteAllResults).toBeCalled();

    expect(spyMessage).toBeCalledWith('成功清空');
    expect(fetchResults).toBeCalled();
  });

  it("should call 'handleMultiDelete' when deleteMore button clicked and ResultsCard 'onSelect'", () => {
    const { wrapper } = setup();
    const instance = wrapper.children().instance();
    const spyHandleDelete = jest.spyOn(instance, 'handleDelete');

    wrapper.find('mock-ResultsCard').first().props().onSelect('1');

    expect(instance.state.selectedKeys.length).toBe(1);

    wrapper.find('.app-page__actions').children().at(2).simulate('click');

    expect(spyHandleDelete).toBeCalledWith(['1']);
  });
});
