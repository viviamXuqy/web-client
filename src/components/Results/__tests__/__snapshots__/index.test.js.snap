// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results component should render right thing and don't change unexpected 1`] = `
Array [
  <mock-Sider
    selectedKey="results-resultsList"
/>,
  <div
    className="app-page"
>
    <div
        className="ant-card app-card"
    >
        <div
            className="ant-card-body"
            style={Object {}}
        >
            <h1
                className="app-page__title"
            >
                结果列表
            </h1>
            <div>
                <form
                    className="ant-form ant-form-inline app-page__filter"
                    onSubmit={[Function]}
                >
                    <div
                        className="fl"
                    >
                        <div
                            className="ant-row ant-form-item"
                            style={undefined}
                        >
                            <div
                                className="ant-form-item-control-wrapper"
                            >
                                <div
                                    className="ant-form-item-control"
                                >
                                    <span
                                        className="ant-form-item-children"
                                    >
                                        <input
                                            className="ant-input result-input"
                                            data-__field={
                                                Object {
                                                    "name": "search",
                                                    "value": "",
                                                  }
                                            }
                                            data-__meta={
                                                Object {
                                                    "name": "search",
                                                    "originalProps": Object {
                                                      "className": "result-input",
                                                      "disabled": false,
                                                      "placeholder": "关键字",
                                                      "prefixCls": "ant-input",
                                                      "type": "text",
                                                    },
                                                    "ref": null,
                                                    "trigger": "onChange",
                                                    "validate": Array [],
                                                    "valuePropName": "value",
                                                  }
                                            }
                                            disabled={false}
                                            id="search"
                                            onChange={[Function]}
                                            onKeyDown={[Function]}
                                            placeholder="关键字"
                                            type="text"
                                            value=""
                                        />
                                    </span>
                                </div>
                            </div>
                        </div>
                        <div
                            className="ant-row ant-form-item"
                            style={undefined}
                        >
                            <div
                                className="ant-form-item-control-wrapper"
                            >
                                <div
                                    className="ant-form-item-control has-success"
                                >
                                    <span
                                        className="ant-form-item-children"
                                    >
                                        <span
                                            className="ant-calendar-picker"
                                            id="range"
                                            onBlur={[Function]}
                                            onFocus={[Function]}
                                            style={
                                                Object {
                                                    "width": 330,
                                                  }
                                            }
                                            tabIndex={0}
                                        >
                                            <span
                                                className="ant-calendar-picker-input ant-input"
                                                onBlur={undefined}
                                                onClick={[Function]}
                                                onContextMenu={undefined}
                                                onFocus={undefined}
                                                onKeyDown={[Function]}
                                                onMouseDown={[Function]}
                                                onMouseEnter={undefined}
                                                onMouseLeave={undefined}
                                                onTouchStart={[Function]}
                                            >
                                                <input
                                                    className="ant-calendar-range-picker-input"
                                                    disabled={undefined}
                                                    placeholder="开始日期"
                                                    readOnly={true}
                                                    tabIndex={-1}
                                                    value=""
                                                />
                                                <span
                                                    className="ant-calendar-range-picker-separator"
                                                >
                                                     ~ 
                                                </span>
                                                <input
                                                    className="ant-calendar-range-picker-input"
                                                    disabled={undefined}
                                                    placeholder="结束日期"
                                                    readOnly={true}
                                                    tabIndex={-1}
                                                    value=""
                                                />
                                                <span
                                                    className="ant-calendar-picker-icon"
                                                />
                                            </span>
                                        </span>
                                    </span>
                                </div>
                            </div>
                        </div>
                        <div
                            className="ant-row ant-form-item"
                            style={undefined}
                        >
                            <div
                                className="ant-form-item-control-wrapper"
                            >
                                <div
                                    className="ant-form-item-control"
                                >
                                    <span
                                        className="ant-form-item-children"
                                    >
                                        <input
                                            className="ant-input result-input"
                                            data-__field={
                                                Object {
                                                    "name": "taskBeginId",
                                                    "value": undefined,
                                                  }
                                            }
                                            data-__meta={
                                                Object {
                                                    "name": "taskBeginId",
                                                    "originalProps": Object {
                                                      "className": "result-input",
                                                      "disabled": false,
                                                      "placeholder": "开始任务ID",
                                                      "prefixCls": "ant-input",
                                                      "type": "text",
                                                    },
                                                    "ref": null,
                                                    "trigger": "onChange",
                                                    "validate": Array [],
                                                    "valuePropName": "value",
                                                  }
                                            }
                                            disabled={false}
                                            id="taskBeginId"
                                            onChange={[Function]}
                                            onKeyDown={[Function]}
                                            placeholder="开始任务ID"
                                            type="text"
                                            value=""
                                        />
                                    </span>
                                </div>
                            </div>
                        </div>
                        <div
                            className="results-list__lable"
                        >
                            至
                        </div>
                        <div
                            className="ant-row ant-form-item"
                            style={undefined}
                        >
                            <div
                                className="ant-form-item-control-wrapper"
                            >
                                <div
                                    className="ant-form-item-control"
                                >
                                    <span
                                        className="ant-form-item-children"
                                    >
                                        <input
                                            className="ant-input result-input"
                                            data-__field={
                                                Object {
                                                    "name": "taskEndId",
                                                    "value": undefined,
                                                  }
                                            }
                                            data-__meta={
                                                Object {
                                                    "name": "taskEndId",
                                                    "originalProps": Object {
                                                      "className": "result-input",
                                                      "disabled": false,
                                                      "placeholder": "结束任务ID",
                                                      "prefixCls": "ant-input",
                                                      "type": "text",
                                                    },
                                                    "ref": null,
                                                    "trigger": "onChange",
                                                    "validate": Array [],
                                                    "valuePropName": "value",
                                                  }
                                            }
                                            disabled={false}
                                            id="taskEndId"
                                            onChange={[Function]}
                                            onKeyDown={[Function]}
                                            placeholder="结束任务ID"
                                            type="text"
                                            value=""
                                        />
                                    </span>
                                </div>
                            </div>
                        </div>
                        <div
                            className="ant-row ant-form-item buttons-group"
                            style={undefined}
                        >
                            <div
                                className="ant-form-item-control-wrapper"
                            >
                                <div
                                    className="ant-form-item-control"
                                >
                                    <span
                                        className="ant-form-item-children"
                                    >
                                        <button
                                            className="ant-btn ant-btn-primary"
                                            onClick={[Function]}
                                            type="submit"
                                        >
                                            <i
                                                className="anticon anticon-search"
                                            />
                                            <span>
                                                检索
                                            </span>
                                        </button>
                                    </span>
                                </div>
                            </div>
                        </div>
                    </div>
                </form>
            </div>
            <div
                className="app-page__actions buttons-group"
            >
                <button
                    className="ant-btn ant-btn-primary"
                    onClick={[Function]}
                    type="button"
                >
                    <i
                        className="anticon iconfont icon-structuredsearc"
                    />
                    <span>
                        结构化检索
                    </span>
                </button>
                <button
                    className="ant-btn ant-btn-primary"
                    onClick={[Function]}
                    type="button"
                >
                    <i
                        className="anticon iconfont icon-search"
                    />
                    <span>
                        以图搜图
                    </span>
                </button>
                <button
                    className="ant-btn ant-btn-warn"
                    disabled={true}
                    onClick={[Function]}
                    type="button"
                >
                    <i
                        className="anticon anticon-primary"
                    />
                    <i
                        className="anticon iconfont icon-delete"
                    />
                    <span>
                        批量删除
                    </span>
                </button>
                <button
                    className="ant-btn ant-btn-danger"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                >
                    <i
                        className="anticon anticon-primary"
                    />
                    <i
                        className="anticon iconfont icon-delete"
                    />
                    <span>
                        清空列表
                    </span>
                </button>
                <button
                    className="ant-btn ant-btn-primary"
                    onClick={[Function]}
                    style={
                        Object {
                            "display": "none",
                          }
                    }
                    type="button"
                >
                    <i
                        className="anticon iconfont icon-therefresh"
                    />
                    <span>
                        重置
                    </span>
                </button>
            </div>
            <div>
                <div
                    className="ant-list results-list ant-list-split ant-list-grid ant-list-something-after-last-item"
                >
                    <div
                        className="ant-spin-nested-loading"
                        style={null}
                    >
                        <div
                            className="ant-spin-container"
                        >
                            <div
                                className="ant-row"
                                style={
                                    Object {
                                        "marginLeft": -10,
                                        "marginRight": -10,
                                      }
                                }
                            >
                                <div
                                    className="ant-col-6"
                                >
                                    <div
                                        className="ant-list-item results-list__item"
                                        style={
                                            Object {
                                                "paddingLeft": 10,
                                                "paddingRight": 10,
                                              }
                                        }
                                    >
                                        <div
                                            className="ant-list-item-content ant-list-item-content-single"
                                        >
                                            <mock-ResultsCard
                                                isResultList={true}
                                                isSelected={false}
                                                onDelete={[Function]}
                                                onSelect={[Function]}
                                                result={
                                                    Object {
                                                        "_id": 0,
                                                        "id": "0",
                                                        "key": "0",
                                                        "targetType": "car",
                                                        "taskId": 0,
                                                        "taskName": "test",
                                                        "taskType": 1,
                                                        "time": 1529355966000,
                                                      }
                                                }
                                                showResultDetail={[Function]}
                                            />
                                        </div>
                                    </div>
                                </div>
                                <div
                                    className="ant-col-6"
                                >
                                    <div
                                        className="ant-list-item results-list__item"
                                        style={
                                            Object {
                                                "paddingLeft": 10,
                                                "paddingRight": 10,
                                              }
                                        }
                                    >
                                        <div
                                            className="ant-list-item-content ant-list-item-content-single"
                                        >
                                            <mock-ResultsCard
                                                isResultList={true}
                                                isSelected={false}
                                                onDelete={[Function]}
                                                onSelect={[Function]}
                                                result={
                                                    Object {
                                                        "_id": 1,
                                                        "id": "1",
                                                        "key": "1",
                                                        "targetType": "car",
                                                        "taskId": 1,
                                                        "taskName": "test",
                                                        "taskType": 1,
                                                        "time": 1529355966000,
                                                      }
                                                }
                                                showResultDetail={[Function]}
                                            />
                                        </div>
                                    </div>
                                </div>
                                <div
                                    className="ant-col-6"
                                >
                                    <div
                                        className="ant-list-item results-list__item"
                                        style={
                                            Object {
                                                "paddingLeft": 10,
                                                "paddingRight": 10,
                                              }
                                        }
                                    >
                                        <div
                                            className="ant-list-item-content ant-list-item-content-single"
                                        >
                                            <mock-ResultsCard
                                                isResultList={true}
                                                isSelected={false}
                                                onDelete={[Function]}
                                                onSelect={[Function]}
                                                result={
                                                    Object {
                                                        "_id": 2,
                                                        "id": "2",
                                                        "key": "2",
                                                        "targetType": "car",
                                                        "taskId": 2,
                                                        "taskName": "test",
                                                        "taskType": 1,
                                                        "time": 1529355966000,
                                                      }
                                                }
                                                showResultDetail={[Function]}
                                            />
                                        </div>
                                    </div>
                                </div>
                                <div
                                    className="ant-col-6"
                                >
                                    <div
                                        className="ant-list-item results-list__item"
                                        style={
                                            Object {
                                                "paddingLeft": 10,
                                                "paddingRight": 10,
                                              }
                                        }
                                    >
                                        <div
                                            className="ant-list-item-content ant-list-item-content-single"
                                        >
                                            <mock-ResultsCard
                                                isResultList={true}
                                                isSelected={false}
                                                onDelete={[Function]}
                                                onSelect={[Function]}
                                                result={
                                                    Object {
                                                        "_id": 3,
                                                        "id": "3",
                                                        "key": "3",
                                                        "targetType": "car",
                                                        "taskId": 3,
                                                        "taskName": "test",
                                                        "taskType": 1,
                                                        "time": 1529355966000,
                                                      }
                                                }
                                                showResultDetail={[Function]}
                                            />
                                        </div>
                                    </div>
                                </div>
                                <div
                                    className="ant-col-6"
                                >
                                    <div
                                        className="ant-list-item results-list__item"
                                        style={
                                            Object {
                                                "paddingLeft": 10,
                                                "paddingRight": 10,
                                              }
                                        }
                                    >
                                        <div
                                            className="ant-list-item-content ant-list-item-content-single"
                                        >
                                            <mock-ResultsCard
                                                isResultList={true}
                                                isSelected={false}
                                                onDelete={[Function]}
                                                onSelect={[Function]}
                                                result={
                                                    Object {
                                                        "_id": 4,
                                                        "id": "4",
                                                        "key": "4",
                                                        "targetType": "car",
                                                        "taskId": 4,
                                                        "taskName": "test",
                                                        "taskType": 1,
                                                        "time": 1529355966000,
                                                      }
                                                }
                                                showResultDetail={[Function]}
                                            />
                                        </div>
                                    </div>
                                </div>
                            </div>
                        </div>
                    </div>
                    <div
                        className="ant-list-pagination"
                    >
                        <ul
                            className="ant-pagination "
                            style={Object {}}
                            unselectable="unselectable"
                        >
                            <li
                                aria-disabled={true}
                                className="ant-pagination-disabled ant-pagination-prev"
                                onClick={[Function]}
                                onKeyPress={[Function]}
                                tabIndex={null}
                                title="上一页"
                            >
                                <a
                                    className="ant-pagination-item-link"
                                />
                            </li>
                            <li
                                className="ant-pagination-item ant-pagination-item-1 ant-pagination-item-active"
                                onClick={[Function]}
                                onKeyPress={[Function]}
                                tabIndex="0"
                                title={1}
                            >
                                <a>
                                    1
                                </a>
                            </li>
                            <li
                                className="ant-pagination-item ant-pagination-item-2"
                                onClick={[Function]}
                                onKeyPress={[Function]}
                                tabIndex="0"
                                title={2}
                            >
                                <a>
                                    2
                                </a>
                            </li>
                            <li
                                className="ant-pagination-item ant-pagination-item-3"
                                onClick={[Function]}
                                onKeyPress={[Function]}
                                tabIndex="0"
                                title={3}
                            >
                                <a>
                                    3
                                </a>
                            </li>
                            <li
                                className="ant-pagination-item ant-pagination-item-4"
                                onClick={[Function]}
                                onKeyPress={[Function]}
                                tabIndex="0"
                                title={4}
                            >
                                <a>
                                    4
                                </a>
                            </li>
                            <li
                                className="ant-pagination-item ant-pagination-item-5 ant-pagination-item-before-jump-next"
                                onClick={[Function]}
                                onKeyPress={[Function]}
                                tabIndex="0"
                                title={5}
                            >
                                <a>
                                    5
                                </a>
                            </li>
                            <li
                                className="ant-pagination-jump-next"
                                onClick={[Function]}
                                onKeyPress={[Function]}
                                tabIndex="0"
                                title="向后 5 页"
                            >
                                <a
                                    className="ant-pagination-item-link"
                                />
                            </li>
                            <li
                                className="ant-pagination-item ant-pagination-item-20"
                                onClick={[Function]}
                                onKeyPress={[Function]}
                                tabIndex="0"
                                title={20}
                            >
                                <a>
                                    20
                                </a>
                            </li>
                            <li
                                aria-disabled={false}
                                className=" ant-pagination-next"
                                onClick={[Function]}
                                onKeyPress={[Function]}
                                tabIndex={0}
                                title="下一页"
                            >
                                <a
                                    className="ant-pagination-item-link"
                                />
                            </li>
                        </ul>
                    </div>
                </div>
            </div>
        </div>
    </div>
</div>,
]
`;
